code/concierge/src/main/kotlin/com/adobe/marketing/mobile/concierge/utils/image/ImageDownloader.kt

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.concierge.utils.image
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.util.LruCache
+import com.adobe.marketing.mobile.concierge.ConciergeConstants
+import com.adobe.marketing.mobile.services.HttpConnecting
+import com.adobe.marketing.mobile.services.HttpMethod
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.NetworkCallback
+import com.adobe.marketing.mobile.services.NetworkRequest
+import com.adobe.marketing.mobile.services.ServiceProvider
+import kotlinx.coroutines.suspendCancellableCoroutine
+import java.io.IOException
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+/**
+ * Image downloader class that handles downloading and caching images in-memory.
+ */
+internal class ImageDownloader {
+
+    companion object {
+        @Volatile
+        private var INSTANCE: ImageDownloader? = null
+
+        fun getInstance(): ImageDownloader {
+            return INSTANCE ?: synchronized(this) {
+                INSTANCE ?: ImageDownloader().also { INSTANCE = it }
+            }
+        }
+    }
+
+    private val cache: LruCache<String, Bitmap>
+
+    init {
+        val maxMemory = Runtime.getRuntime().maxMemory()
+        val calculatedCacheSize = (maxMemory / 4).toInt() // Use 1/4th of available memory for cache
+        val minCacheSize = 50 * 1024 * 1024 // 50MB cache
```

Comment:
Let us use a much smaller size like 5MB to start with and hardcode it instead if accessing Runtime.

Reply:
the LruCache is now using the maxEntries param being passed in to the DefaultImageProvider when it is initialized. for concierge purposes, do we want to use the max entries (which is currently 64) or do we want to set a maximum cache size?